### PR TITLE
Moving --lock-file, --seccomp-profile-root and --node-status-max-images Kubelet flags to configuration

### PR DESCRIPTION
**Pull request by knabben**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:

/kind cleanup
/area kubelet
/area kubelet-api

**What this PR does / why we need it**:

Moving --lock-file, --exit-on-lock-contention, --seccomp-profile-root, --node-status-max-images to Kubelet configuration

**Which issue(s) this PR fixes**:

Refs https://github.com/kubernetes/kubernetes/issues/86843

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
The Kubelet's `--lock-file`, `--exit-on-lock-contention`, `--seccomp-profile-root`, `--node-status-max-images` option is now available via the Kubelet config file field `lockFilePath`, `exitOnLockContention`, `seccompProfileRoot` and `nodeStatusMaxImage` respectively.
```

### CODE REVIEW

File: pkg/kubelet/apis/config/types.go [hunk 1]
```diff
@@ -347,6 +347,22 @@ type KubeletConfiguration struct {
 	// The purpose of this format is make sure you have the opportunity to notice if the next release hides additional metrics,
 	// rather than being surprised when they are permanently removed in the release after that.
 	ShowHiddenMetricsForVersion string
+
+	/* Experimental flags */
+
+	// lockFilePath is the path that kubelet will use to as a lock file.
+	// It uses this file as a lock to synchronize with other kubelet processes that may be running.
+	LockFilePath string
```
Comment (dashpole) [inline pkg/kubelet/apis/config/types.go:355]: The linked issue says for experimental flags "Below is the list of flags that needs to be evaluated and can be marked into categories like deprecated, removed. The remaining flags can be safely migrated to a stable config file format."

Have these experimental feature flags been evaluated?  NodeStatusMaxImages seems like a reasonable candidate to graduate, but i'm not sure about the others above...

Comment (knabben) [reply pkg/kubelet/apis/config/types.go:355]: What is the better way to assess if the flag is a candidate to graduate? And what makes the flag Alpha yet, besides the *<Warning: Alpha feature>* in the description.

Comment (dashpole) [reply pkg/kubelet/apis/config/types.go:355]: It probably requires digging into the initial reason why they were added, and either asking the people who added the flags, or bringing it up at sig-node.

Comment (knabben) [reply pkg/kubelet/apis/config/types.go:355]: alright, fair enough.

**Comment by mtaufen** [comment]
See https://github.com/kubernetes/kubernetes/pull/90745#issuecomment-627656728 - as @dashpole said, we need to do some archaeology before moving these.

**Pull request closed without merging**